### PR TITLE
Initial commit.

### PR DESCRIPTION
**Pull request by DanielaS12**
Repository for logging the armory legal notice.

More details:

https://www.notion.so/Plugins-legal-notice-2d87fb0ba855453a9e6797c3d1240046

https://armory.atlassian.net/browse/SPLAT-597

### CODE REVIEW

File: src/main/java/io/armory/spinnaker/legalnotice/Config.java [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
 
 import lombok.extern.slf4j.Slf4j;
+import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
 @Configuration
 @Slf4j
 public class Config {
```
Comment (ovidiupopa07) [inline src/main/java/io/armory/spinnaker/legalnotice/Config.java:11]: should be an auto-configured class that it is automatically loaded if the jar is added to the plugin classpath

Comment (ovidiupopa07) [reply src/main/java/io/armory/spinnaker/legalnotice/Config.java:11]: with the current impl the bean will not be registered unless the package is added to the component scan classpath

**Review by ovidiupopa07 — CHANGES_REQUESTED** [review]

File: .github/workflows/auto_approve_astrolabe_prs.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: Auto approve astrolabe pr
```
Comment (ovidiupopa07) [inline .github/workflows/auto_approve_astrolabe_prs.yml]: Astrolabe is not required here

File: .github/workflows/build_cd_artifacts.yml [hunk 3]
```diff
@@ -29,45 +24,36 @@ jobs:
...
-          echo 'registry=https://armory.jfrog.io/artifactory/api/npm/npm-all/' > ~/.npmrc
-          curl -u${ARTIFACTORY_USER}:${ARTIFACTORY_PASSWORD} https://armory.jfrog.io/artifactory/api/npm/auth >> ~/.npmrc
-
-      - name: Build
-        run: |
-          yarn --registry https://armory.jfrog.io/artifactory/api/npm/npm-all/ --frozen-lockfile
-          yarn test
-          yarn build --deck-version ${{ steps.variables.outputs.version }}
 
       - name: Build Image
```
Comment (ovidiupopa07) [inline .github/workflows/build_cd_artifacts.yml]: no need to build a docker image. This is a plain jar artifact

File: .github/workflows/build_cd_artifacts.yml [hunk 4]
```diff
@@ -29,45 +24,36 @@ jobs:
...
-          docker build -t ${{ steps.variables.outputs.artifactory_image_name }} .
+          ./gradlew docker \
+            -PcdBuild=true \
+            -PimageName=${{ steps.variables.outputs.artifactory_image_name }} \
+            -Pversion=${{ steps.variables.outputs.version }} \
+            -Partifactory_user=${ARTIFACTORY_USER} \
+            -Partifactory_password=${ARTIFACTORY_PASSWORD} \
+            -Prelease.version=${{ steps.variables.outputs.version_as_semver }}
 
       - name: Publish Image To Artifactory
```
Comment (ovidiupopa07) [inline .github/workflows/build_cd_artifacts.yml]: no need to publish the image

File: .github/workflows/update_base_service.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
+name: Update Base Service
```
Comment (ovidiupopa07) [inline .github/workflows/update_base_service.yml]: no need for this workflow

File: .github/workflows/update_extension_dependency.yml [hunk 6]
```diff
@@ -0,0 +1,32 @@
+name: Update Extension Dependency
```
Comment (ovidiupopa07) [inline .github/workflows/update_extension_dependency.yml]: no need for this one

**Review by ovidiupopa07 — CHANGES_REQUESTED** [review]
The java implementation looks good but there is still work to do in the github actions. I see no step the uploads the jar file to artifactory

File: .github/workflows/build-release.yml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 17
+
+      - name: Login to Artifactory
+        run: |
+          docker login -u ${{secrets.ARTIFACTORY_USER}} -p ${{secrets.ARTIFACTORY_TOKEN}} armory-docker-local.jfrog.io
+        shell: bash
+
+      - name: Publish To Artifactory
```
Comment (ovidiupopa07) [inline .github/workflows/build-release.yml]: I think this one is pushing the docker image 
`  ./gradlew --no-build-cache build -x check dockerPush $GRADLE_ARGS`

also it has references to `armory-enterprise-console` which should not happen

**Review by ovidiupopa07 — CHANGES_REQUESTED** [review]

**Comment by ovidiupopa07** [comment]
hi @DanielaS12 - one thing to remember after we merge this PR. We should add the jar file to the pipeline-as-a-code plugin project and check (in an integrated way) that the log is visible.

File: .github/workflows/build-release.yml [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+          ./gradlew --no-build-cache build -x check dockerPush $GRADLE_ARGS
+          echo "BUILD_NUMBER=$(./gradlew $GRADLE_ARGS  printVersionNumber|grep VERSION|awk -F: '{print $2}')" >> $GITHUB_ENV
+
+      - name: Publish Core To Artifactory
+        id: publish_core
+        run: |
+          echo "BUILD_NUMBER=$(./gradlew $GRADLE_ARGS  printVersionNumber|grep VERSION|awk -F: '{print $2}')" >> $GITHUB_ENV
+          ./gradlew --no-build-cache build artifactoryPublish $GRADLE_ARGS
+
+      - name: Scan Using Aquasec
```
Comment (ovidiupopa07) [inline .github/workflows/build-release.yml]: not needed since this one is scanning a docker image

**Review by ovidiupopa07 — CHANGES_REQUESTED** [review]

File: .github/workflows/build-release.yml (hunk 7 above)
Comment (DanielaS12) [reply .github/workflows/build-release.yml]: removing all docker related actions

**Review by ovidiupopa07 — APPROVED** [review]
Looks ok. Let's give it a go and see if the jar is published

**Pull request merged**